### ISSUE TITLE
To use trigger when calling stored procedure

### ISSUE THREAD

**Issue by abhijaykh07** [Azure/azure-cosmosdb-js-server#44]
I have a scenario where I am calling stored procedure for bulkImport. So is there any option to call trigger from stored procedure while creating document ?
In trigger there is a pre logic that needs to be executed before document.

**Comment by abhijaykh07** [comment]
@mkolt Can you please suggest me solution for this issue ?

**Comment by mkolt** [comment]
calling triggers are not supported for operations within script (from sproc, in particular). Since sproc already runs in JavaScript, just execute your pre-logic and then call getContext().getCollection().createDocument(...), adding support for triggers wouldn't add much functionality, just convenience. Does this make sense?

**Comment by abhijaykh07** [comment]
> calling triggers are not supported for operations within script (from sproc, in particular). Since sproc already runs in JavaScript, just execute your pre-logic and then call getContext().getCollection().createDocument(...), adding support for triggers wouldn't add much functionality, just convenience. Does this make sense?

@mkolt 
yes that's the option left as of now, writing the trigger logic before create document in the same stored procedure.

I have also tried while calling from c# end, passing via PreTriggerInclude. But its giving me always an exception "one of the specified inputs is invalid". Do you have any idea for this issue ?
But stored procedure works if I don't pass PreTriggerInclude option.

![image](https://user-images.githubusercontent.com/60259083/73133635-a865e480-4051-11ea-8ae9-6b85b71f619e.png)
